endpoints/api/user.py

```diff
@@ -545,6 +546,83 @@ def delete(self):
         return "", 204
 
 
+@resource("/v1/user/initialize")
+class InitializeUser(ApiResource):
+
+    schemas = {
+        "InitializeUser": {
+            "type": "object",
+            "description": "Fields which must be specified for a new user.",
+            "required": [
+                "username",
+                "password",
+            ],
+            "properties": {
+                "username": {
+                    "type": "string",
+                    "description": "The user's username",
+                },
+                "password": {
+                    "type": "string",
+                    "description": "The user's password",
+                },
+                "email": {
+                    "type": "string",
+                    "description": "The user's email address",
+                },
+            },
+        },
+    }
+
+    @verify_not_prod
+    @nickname("initializeUser")
+    @internal_only
+    @validate_json_request("InitializeUser")
+    @anon_allowed
+    def post(self):
+        """
+        Create initial user in an empty database
+        """
+
+        # Ensure that we are using database auth.
+        if app.config["AUTHENTICATION_TYPE"] != "Database":
+            raise request_error(message="Cannot initialize user in a non-database auth system")
+
+        if model.user.get_active_user_count() != 0:
+            raise request_error(message="Cannot initialize user in a non-empty database")
+
+        user_data = request.get_json()
+        try:
+            prompts = model.user.get_default_user_prompts(features)
+            new_user = model.user.create_user(
+                user_data["username"],
+                user_data["password"],
+                user_data.get("email"),
+                auto_verify=True,
+                email_required=features.MAILING,
+                is_possible_abuser=False,
+                prompts=prompts,
+            )
+            success, headers = common_login(new_user.uuid)
+            if not success:
+                raise request_error(message="Could not login. Failed to initialize user")
+
+            return (
+                {
+                    "username": user_data["username"],
+                    "email": user_data.get("email"),
+                    "password": user_data["password"],
+                    "encrypted_password": authentication.encrypt_user_password(
```

Comment:
**Don't return passwords**

Reply:
I'll need the `encrypted_password` returned for use in follow-up API calls (that's the bearer token, right?). I can remove the `password`.

Reply:
No. Passwords and tokens are different.